pkg/listener/http.go

```diff
@@ -122,24 +136,49 @@ func (l *HttpListener) Close() {
 	}
 }
 
-func (l *HttpListener) doPreFilter(ctx *fasthttp.RequestCtx) error {
+func (l *HttpListener) doPreFilter(ctx context.Context, fastHttpCtx *fasthttp.RequestCtx) error {
+	newCtx, span := tracing.GetTraceSpan(ctx, "http_listener_do_pre_filter")
+	defer span.End()
 	for i := 0; i < len(l.preFilters); i++ {
 		f := l.preFilters[i]
-		err := f.PreHandle(ctx)
+		err := f.PreHandle(newCtx, fastHttpCtx)
 		if err != nil {
+			tracing.RecordErrorSpan(span, err)
 			return err
 		}
 	}
 	return nil
 }
 
-func (l *HttpListener) doPostFilter(ctx *fasthttp.RequestCtx) error {
+func (l *HttpListener) doPostFilter(ctx context.Context, fastHttpCtx *fasthttp.RequestCtx) error {
+	newCtx, span := tracing.GetTraceSpan(ctx, "http_listener_do_post_filter")
+	defer span.End()
```

Comment:
I recommend not adding `TraceContext` for this location, because there is no very complicated logic here.